modules/social_features/social_follow_user/src/Form/SocialFollowUserSettingsForm.php

```diff
@@ -52,8 +54,8 @@ public function submitForm(array &$form, FormStateInterface $form_state): void {
     // Remove the permissions for following users if disabled.
     if ($follow_status === TRUE && $form_state->getValue('status') === 0) {
       // Permission can be different from default so retrieve it dynamically.
-      $roles = user_role_names(FALSE, $permission);
-      $roles = array_keys($roles);
+      $entity_roles = array_filter(Role::loadMultiple(), fn(RoleInterface $role) => $role->hasPermission($permission));
```

Comment:
Can you use dependency injection instead of using the entity directly ? Eg. $this->entityTypeManager->getStorage('user_role')->loadMultiple()